packages/vaadin-grid/test/keyboard-navigation.test.js

```diff
@@ -1956,17 +1956,47 @@ describe('keyboard navigation', () => {
 
       const spy = sinon.spy();
 
-      grid.addEventListener('cell-focus', (e) => {
+      const eventHandler = (e) => {
         spy();
 
         const context = e.target.getEventContext(e);
         expect(context).to.deep.equal(expectedContext);
         done();
-      });
+      };
+
+      grid.addEventListener('cell-focus', eventHandler);
 
       left();
 
       expect(spy.calledOnce).to.be.true;
+      grid.removeEventListener('cell-focus', eventHandler);
+    });
+
+    it('should contain event context in event detail', () => {
+      tabToBody();
+      right();
+
+      const spy = sinon.spy();
+
+      grid.addEventListener('cell-focus', spy);
+
+      left();
+
+      const e = spy.firstCall.args[0];
+
+      expect(e.detail.context).to.be.deep.equal({
+        column: grid.querySelector('vaadin-grid-column'),
+        detailsOpened: false,
+        expanded: false,
+        index: 0,
+        item: 'foo',
+        level: 0,
+        section: 'body',
+        selected: false
+      });
+      expect(spy.callCount).to.be.equal(1);
```

Comment:
The spy's call count should probably be asserted before asserting the spy's arguments. Otherwise, it won't be reached when `spy.callCount === 0`.